smoketests/tests/replication.py

```diff
@@ -247,31 +247,42 @@ def start(self, id: int, count: int):
         """Send a message to the database."""
         retry(lambda: self.call("start", id, count))
 
+    def collect_counter_rows(self):
+        return int_vals(self.cluster.sql("select * from counter"))
+
+
 class LeaderElection(ReplicationTest):
     def test_leader_election_in_loop(self):
         """This test fails a leader, wait for new leader to be elected and verify if commits replicated to new leader"""
         iterations = 5
         row_ids = [101 + i for i in range(iterations * 2)]
         for (first_id, second_id) in zip(row_ids[::2], row_ids[1::2]):
             cur_leader = self.cluster.wait_for_leader_change(None)
+            print(f"ensure leader health {first_id}")
             self.cluster.ensure_leader_health(first_id)
 
-            print("killing current leader: {}", cur_leader)
+            print(f"killing current leader: {cur_leader}")
             container_id = self.cluster.fail_leader()
 
             self.assertIsNotNone(container_id)
 
             next_leader = self.cluster.wait_for_leader_change(cur_leader)
             self.assertNotEqual(cur_leader, next_leader)
             # this check if leader election happened
+            print(f"ensure_leader_health {second_id}")
             self.cluster.ensure_leader_health(second_id)
             # restart the old leader, so that we can maintain quorum for next iteration
+            print(f"reconnect leader {container_id}")
             self.cluster.restore_leader(container_id, 'start')
 
-        # verify if all past rows are present in new leader
-        for row_id in row_ids:
-            table = self.sql(f"SELECT * FROM counter WHERE id = {row_id}")
-            self.assertIn(f"{row_id}", str(table))
+        # Ensure we have a current leader
```

Comment:
Not sure exactly why, but test becomes flaky if I omit this (same for `test_leader_c_disconnect_in_loop`).